module/advancement/types/scale-value.mjs

```diff
@@ -2,6 +2,195 @@ import Advancement from "../advancement.mjs";
 import AdvancementConfig from "../advancement-config.mjs";
 import AdvancementFlow from "../advancement-flow.mjs";
 
+/**
+ * Class responsible for formatting and presenting a scale value of a specific type.
+ * @param {ScaleValueAdvancement} advancement  The advancement to which this scale value belongs.
+ * @param {object} data                        Data for a scale value at a specific level.
+ */
+export class ScaleValueType {
+  constructor(advancement, data) {
+    Object.assign(this, data);
+    this.advancement = advancement;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Information on how a scale value of this type is configured.
+   *
+   * @typedef {object} ScaleValueTypeMetadata
+   * @property {string} label       Name of this type.
+   * @property {string} hint        Hint for this type shown in the scale value configuration.
+   * @property {boolean} isNumeric  When using the default editing interface, should numeric inputs be used?
+   */
+
+  /**
+   * Configuration information for this scale value type.
+   * @type {ScaleValueTypeMetadata}
+   */
+  static get metadata() {
+    return {
+      label: "DND5E.AdvancementScaleValueTypeString",
+      hint: "DND5E.AdvancementScaleValueTypeHintString",
+      isNumeric: false
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Attempt to convert another scale value type to this one.
+   * @param {ScaleValueType} original  Original type to attempt to convert.
+   * @returns {object|null}
+   */
+  static converted(original) {
+    return { value: original.prepared };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * This scale value prepared to be used in roll formulas.
+   * @type {string|null}
+   */
+  get prepared() { return this.value; }
```

Comment:
Maybe we could call this `formula` instead? Assuming it's possible to use this value in such a way: `new Roll(scaleValue.formula)` then I think that name would align with the `Roll` parameter name.